### PR TITLE
[11.x] Allow `lcobucci/jwt` v5 and cleaned up version constraints

### PR DESCRIPTION
**Pull request by GrahamCampbell**

### CODE REVIEW

**Comment by GrahamCampbell** [comment]
Ohhhh, man. Looking upstream at https://github.com/thephpleague/oauth2-server, things seem to be in a right mess. They say they support PHP 7.2, but actually the depednencies are not installable on that version, the tests are failing across the board, and PHP 8.1 and 8.2 are untested. I wonder if they need a little help.

File: composer.json [hunk 1]
```diff
@@ -26,7 +26,7 @@
         "illuminate/encryption": "^9.0|^10.0",
         "illuminate/http": "^9.0|^10.0",
         "illuminate/support": "^9.0|^10.0",
-        "lcobucci/jwt": "^3.4|^4.0",
+        "lcobucci/jwt": "^5.0",
```
Comment (driesvints) [inline composer.json:29]: We can't drop the old versions as Passport still supports PHP 8.0

Comment (GrahamCampbell) [reply composer.json:29]: I know. this is a temporary commit to force the CI to install v5, so I could see the conflicts.

**Comment by Sephster** [comment]
Should be releasing a new version in the next day or so to fix PHP version issues. Apologies for the delay

**Comment by Sephster** [comment]
That's version 8.5.0 tagged

**Comment by GrahamCampbell** [comment]
@Sephster Any chance of a 8.5.1 tag please?

**Comment by Sephster** [comment]
Will tag in the next 30 mins

**Comment by Sephster** [comment]
Tagged @GrahamCampbell

**Comment by GrahamCampbell** [comment]
@driesvints ready for review now. 🎉

**Review by driesvints — APPROVED** [review]
Lgtm. Thanks @GrahamCampbell @Sephster

**Comment by taylorotwell** [comment]
Thanks all.

**Pull request merged**